src/sql/src/plan/query.rs

```diff
@@ -701,36 +702,71 @@ pub fn plan_mutation_query(
         bail!("cannot mutate system table '{}'", table.name());
     }
 
-    let get = HirRelationExpr::Get {
+    let mut relation_expr = HirRelationExpr::Get {
         id: Id::Global(table.id()),
         typ: desc.typ().clone(),
     };
 
-    let selection: HirRelationExpr = match selection {
-        Some(mut expr) => {
-            transform_ast::transform_expr(scx, &mut expr)?;
-            let expr = resolve_names_expr(&mut qcx, expr)?;
-            let ecx = &ExprContext {
-                qcx: &qcx,
-                name: "WHERE clause",
-                scope: &Scope::from_source(
-                    Some(PartialName::from(table.name().clone())),
-                    desc.iter_names(),
-                    None,
-                ),
-                relation_type: &RelationType::new(desc.iter_types().cloned().collect()),
-                allow_aggregates: false,
-                allow_subqueries: true,
+    let relation_type = qcx.relation_type(&relation_expr);
+
+    let scope = Scope::from_source(
+        Some(PartialName::from(table.name().clone())),
+        desc.iter_names(),
+        None,
+    );
+
+    if let Some(mut expr) = selection {
+        // If `DELETE FROM <table_name> USING <using> WHERE <selection>`...
+        if !using.is_empty() {
```

Comment:
Oh, this is actually better than what I had in mind for the `DELETE` without `USING` case. I was thinking we could just use a `Constant ()` operator for the join within the `EXISTS` subquery... but not generating the subquery at all is much better.

Reply:
Another thing that it is better than what I had in mind is that I was thinking we would manually generate here the `Hir` tree, instead of the AST and then plan it, which would force us to deal with the name resolution scopes manually, which is always an unpleasant experience.